src/ecs/Engine.hpp

```diff
@@ -0,0 +1,137 @@
+#include "systems/Movement.hpp"
+#include "systems/PositionLogger.hpp"
+
+namespace ecs
+{
+    /**
+     * The Engine class manages the whole part of systems in ecs in a given world.
+     * Systems are, most of the time, short function using components to realise some actions.
+     * For example, to make an Entity move, we need a component Position and a component Velocity,
+     * the system is the thing who update the position of the Entity according to the Velocity component
+     * by adding velocity value to the x & y member of the component position of the Entity.
+     */
+    class Engine {
+      public:
+        /**
+         * Constructor of Engine class, it takes care of register every component in the world's registry.
+         * It also create a player and an enemy.
+         * And finally it takes care of adding every systems related to the components in the initWorld.
+         * @param wSizeWidth width size of the window in pixel, 800px as default
+         * @param wSizeHeight height size of the window in pixel, 600 as default
+         * @param wTitle window's title, "r-type" as default
+         */
+        explicit Engine(int wSizeWidth = 800, int wSizeHeight = 600, std::string wTitle = "r-type")
+            : _worldSwitchReady(false)
+        {
+            _window = std::make_unique<sf::RenderWindow>(sf::VideoMode(wSizeWidth, wSizeHeight), wTitle);
+            _window.get()->setFramerateLimit(60);
+            ecs::World initWorld(_window);
+            ecs::Entity player = initWorld.registry.spawn_entity();
+            initWorld.registry.addComponent<ecs::component::Position>(player, {10, 10});
+            initWorld.registry.addComponent<ecs::component::Velocity>(player, {5, 5});
+            initWorld.registry.addComponent<ecs::component::Size>(player, {2, 2});
+            initWorld.registry.addComponent<ecs::component::Direction>(player, {0, 0});
+            initWorld.registry.addComponent<ecs::component::Drawable>(
+                player, {"src/demo/assets/textures/players.gif", {1, 1, 32, 16}});
+            initWorld.registry.addComponent<ecs::component::Controllable>(
+                player, {sf::Keyboard::Z, sf::Keyboard::Q, sf::Keyboard::S, sf::Keyboard::D});
+
+            ecs::Entity enemy = initWorld.registry.spawn_entity();
+            initWorld.registry.addComponent<ecs::component::Position>(enemy, {500, 500});
+            initWorld.registry.addComponent<ecs::component::Size>(enemy, {5, 5});
+            initWorld.registry.addComponent<ecs::component::Drawable>(
+                enemy, {"src/demo/assets/textures/players.gif", {1, 18, 32, 16}});
+            initWorld.registry.addComponent<ecs::component::EnemyAI>(enemy, {});
+
+            initWorld.addSystem(ecs::systems::handleSFMLEvents);
+            initWorld.addSystem(ecs::systems::handleSFMLMovements);
+            initWorld.addSystem(ecs::systems::manageClientEvents);
+            // initWorld.addSystem(ecs::systems::positionLogger);
```

Comment:
We should consider adding a compiler definitition or some sort of switch to activate when we need to.
How about it @Bootoyka @aurelenc  ?

Reply:
Are u talking about systems ? if yes understand that they are continualy running in the current world